### DISCUSSION TITLE
Rerender a page getstaticprops

### DISCUSSION THREAD

**Discussion by IRediTOTO** [vercel/next.js#14770] [Ideas]
Hi, In my homepage I use getStaticProps to show my products. But if I add more products, it will not show, so I use SWR to fetch new data, it's pretty fine. 
Is there any way or trick to **re-render homepage to add new data**? I don't want fetch api because it look slow.
We can use web hook to rebuild app, but look like its not the best way for just small task.
I want something work like fallback in getStaticPaths

**Comment by eddyw** [comment]
Hi,

NextJS has an experimental feature called "Incremental Static Regeneration" which I think it's what you're looking for. It's documented here:
https://github.com/vercel/next.js/discussions/11552

You basically just need to add `unstable_revalidate: 1` (where `1` means "at most once every 1 second") to `getStaticProps`

**Reply by IRediTOTO** [reply]
Thank you, I got it https://nextjs.org/blog/next-9-4#incremental-static-regeneration-beta
But what is the mean of "at most once every 1 second" ? what if I set unstable_revalidate:100 ? 
If user goto page /post/abc Nextjs will send old version of page, then on server will render new one, and other user will get newest version of page, right? 
I wonder can this make app slow? But look like it work like getServerSideProps...

**Reply by eddyw** [reply]
@warcraft14115 it means that the cache needs to revalidate. If your content doesn't update too often, you can set to a higher value. For instance, `unstable_revalidate: 5 * 60` will re-generate content in the background at most once every 5min. While the content re-generates in the background, the user will get the last cached copy.
